lib/rspec/matchers/built_in/compound.rb

```diff
@@ -36,6 +36,28 @@ def expects_call_stack_jump?
           NestedEvaluator.matcher_expects_call_stack_jump?(matcher_2)
         end
 
+        # @api private
+        # @return [Boolean]
+        def diffable?
+          [matcher_1, matcher_2].any?(&method(:matcher_is_diffable?))
```

Comment:
Looks like this can fail for 1.9.2. So I will make it pass explicit block with method call inside..